File: tiq-e/others/6/main.ts
import {performance} from "perf_hooks";

function missingNumber(nums: number[]): number {
    let n: number = nums.length;
    let result: number = (n * (n + 1)) / 2;
    for (let i = 0; i < n; ++i) {
        result -= nums[i];
    }
    return result;
}

const x = [3, 0, 1];

const start = performance.now();
const result = missingNumber(x);
const end = performance.now();
console.log("Duration:", end - start);
console.log("Result:", result);

File: tiq-e/others/6/main.go
package main

import (
	"fmt"
	"time"
)

func missingNumber(nums []int) int {
	var expected int
	var actual int
	for i := 0; i < len(nums); i++ {
		actual += nums[i]
		expected += i + 1
	}

	return expected - actual
}

func main() {
	x := []int{3, 0, 1}

	start := time.Now()
	result := missingNumber(x)
	end := time.Since(start)
	fmt.Println("Duration: ", end)
	fmt.Println("Result: ", result)
}
